### ISSUE TITLE
Formatting on save should be disabled by default

### ISSUE THREAD

**Issue by EJammy** [LunarVim/LunarVim#2444]
### Feature Description

Formatting on save changes user's file with no easy way to undo. It should be disabled.
## Reasons:
- This is the behavior on most (all?) editors
- Formatting on save might mess up someone's file when they are just trying LunarVim for the first time

I know default configurations are opinions, but I don't think formatting on save is a good thing to enable by default. I installed LunarVim to try it out on my files and had to find ways to restore it because LunarVim formats it unexpectably.

### Describe the alternatives you have considered

```vim
:au! format_on_save
```

**Comment by abzcoding** [comment]
change this line in your `~/.config/lvim/config.lua`
```lua
lvim.format_on_save = true
```
to 
```lua
lvim.format_on_save = false
```

**Comment by abzcoding** [comment]
>  user's file with no easy way to undo

hitting the `u` key in normal mode  will revert the file back

**Comment by abzcoding** [comment]
cc: @kylo252 what do you think?
i have this disabled in my own config as well

**Comment by kylo252** [comment]
> Formatting on save changes user's file with no easy way to undo. It should be disabled.

It should be easy to undo tho, since both `undodir` and `undofile` are set.

> cc: @kylo252 what do you think? i have this disabled in my own config as well

I actually have it set only for `lua` files, so maybe we can/should tweak the default pattern, see `:h autocmd-pattern`

```lua
lvim.format_on_save = {
  ---@usage pattern string pattern used for the autocommand (Default: '*')
  pattern = "*.lua",
  ---@usage timeout number timeout in ms for the format request (Default: 1000)
  timeout = 1000,
}
```

**Comment by abzcoding** [comment]
```lua
lvim.format_on_save = {
  ---@usage pattern string pattern used for the autocommand (Default: '*')
  pattern = "*.lua",
  ---@usage timeout number timeout in ms for the format request (Default: 1000)
  timeout = 1000,
}
```
lets put this in the default config.lua

**Comment by danielo515** [comment]
I think sticking to what most editors do is better. I think most editors do format on save, and I actually like this behaviour.
Why don't you just configure it to your liking?

**Comment by abzcoding** [comment]
> I think sticking to what most editors do is better. I think most editors do format on save, and I actually like this behaviour.
> Why don't you just configure it to your liking?

we should at least use this syntax in 
```lua
lvim.format_on_save = {
  ---@usage pattern string pattern used for the autocommand (e.g: '*.lua')
  pattern = "*",
  ---@usage timeout number timeout in ms for the format request (Default: 1000)
  timeout = 1000,
}
```
instead of
https://github.com/LunarVim/LunarVim/blob/b4d5f093a5607d3f14cbf90aebfb25036c3ee272/utils/installer/config.example.lua#L13-L14

**Comment by danielo515** [comment]
I agree that putting it more accesible on the default config is the best thing to do

**Comment by EJammy** [comment]
> I think sticking to what most editors do is better. I think most editors do format on save, and I actually like this behaviour. Why don't you just configure it to your liking?

I did! My concern is that when I first tested lunarvim on my files, it auto formatted the file and I didn't know how to restore at first (you can't undo and save because saving triggers formatting).

**Comment by Uzaaft** [comment]
The default 

> I think sticking to what most editors do is better. I think most editors do format on save, and I actually like this behaviour.
> Why don't you just configure it to your liking?

The vscode editor does not do this as default.
Nor does other professional IDE's, like clion, pycharm, etc.

**Comment by abzcoding** [comment]
> Regarding what you wrote @abzcoding .
I think that would confuse some users?

probably 😢

**Comment by Uzaaft** [comment]
> > Regarding what you wrote @abzcoding .
> > I think that would confuse some users?
> 
> probably 😢

Not everyone is a beast like you xD
I can add those to Lunarvim.org if you'd like.
I can go ahead change the default value in the example config. I think it makes sense to have this off as default, since most IDE's does that as well.

**Comment by rebuilt** [comment]
It used to be OFF by default.  When it was OFF by default, a lot of users opened issues asking how to turn it on.  Now that the default is ON, there aren't as many issues open for it.  Just giving some history here.

**Comment by rebuilt** [comment]
We started with the autoformat setting as OFF.  We set the example config to ON because we had so many open issues asking how to turn it on.  Now that it's the default option, we don't get as many questions about it.
